homeassistant/components/emoncms/coordinator.py

```diff
@@ -24,8 +23,15 @@ def __init__(
         """Initialize the emoncms data coordinator."""
         super().__init__(
             hass,
-            _LOGGER,
+            LOGGER,
             name="emoncms_coordinator",
-            update_method=emoncms_client.async_list_feeds,
             update_interval=scan_interval,
         )
+        self.emoncms_client = emoncms_client
+
+    async def _async_update_data(self) -> list[dict[str, Any]]:
+        """Fetch data from API endpoint."""
+        data = await self.emoncms_client.async_request("/feed/list.json")
```

Comment:
What's the difference between `emoncms_client.async_list_feeds` and this?

Reply:
`async_list_feeds` just return the list of feeds or an empty list in case the connexion is successfull. If the connection is not successfull it returns `None`

`async_requests` is more generic and it returns the payload where you can have the error message :
```
{"success": True/False, "message": list of feeds / empty list if no feeds / error message}